File: admin/src/main/java/com/project/admin/config/SecurityConfig.java
package com.project.admin.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.web.SecurityFilterChain;

@Configuration
public class SecurityConfig {

    @Bean
    public PasswordEncoder passwordEncoder() {

        return new BCryptPasswordEncoder();
    }

    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
        http.csrf(csrf -> csrf.disable())
                        .authorizeHttpRequests(auth ->
                        auth.requestMatchers("/admin/signup", "/admin/signin").permitAll()
                                .requestMatchers("/admin/**").authenticated()
                ).
                formLogin(form ->
                        form.loginProcessingUrl("/admin/login")
                                .loginPage("/admin/login"))

                .sessionManagement(session -> session
                        .sessionCreationPolicy(SessionCreationPolicy.IF_REQUIRED)
                );

        return http.build();
    }

}

File: api/src/main/kotlin/com/project/api/config/SecurityConfig.kt
package com.project.api.config

import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.web.SecurityFilterChain
import org.springframework.web.cors.CorsConfigurationSource

@Configuration
class SecurityConfig(
    @Qualifier(value = "corsConfigurationSource") private val cors: CorsConfigurationSource,
) {

    @Bean
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
        http
            .cors { it.configurationSource(cors) }
            .csrf { it.disable() }  //TODO 추후 JWT 인증 도입 시 고려해 CSRF 토큰 비활성화 처리
            .authorizeHttpRequests { it.anyRequest().permitAll() }
        return http.build()
    }
}